src/wallet/rpcwallet.cpp

```diff
@@ -4567,6 +4567,59 @@ static RPCHelpMan upgradewallet()
     };
 }
 
+RPCHelpMan analyzerawtransaction()
+{
+    return RPCHelpMan{"analyzerawtransaction",
+        "\nCalculate the balance change resulting in the signing and broadcasting of the given transaction.\n" +
+        HELP_REQUIRING_PASSPHRASE,
+        {
+            {"hexstring", RPCArg::Type::STR, RPCArg::Optional::NO, "The transaction hex string"},
+        },
+        RPCResult{
+            RPCResult::Type::NUM, "", "The wallet balance change (negative means decrease)."
+        },
+        RPCExamples{
+            HelpExampleCli("analyzerawtransaction", "\"myhex\"")
+            + HelpExampleRpc("analyzerawtransaction", "\"myhex\"")
+        },
+    [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
```

Comment:
Nit: People typically place `const`s before types in this codebase, AFAIK. If I'm wrong, please correct me.

Reply:
Looking around in the code, the way I wrote it seems to be the way it's written elsewhere, so keeping as is.

Reply:
Two lines above the `const` modifier is before type. Anyway, it's not important so I don't want to waste your time on this any further.

Reply:
That would actually make the shared pointer a constant, which points to a non-constant CWallet instance.

See e.g. https://stackoverflow.com/questions/7526152/easy-rule-to-read-complicated-const-declarations

Reply:
Also try modifying my PR to this, and try compiling:
```C++
    const std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
    if (!wallet) return NullUniValue;
    wallet->ResendWalletTransactions(); // <- this is not const
```
(it works, even though ResendWalletTransactions() is NOT const-declared)

Reply:
Thanks for the clarification!

Reply:
Actually, I'm mistaken, and the two versions are identical. (ResendWalletTransactions() compiles for both cases.)
I actually think this is a mistake, and could just as well be
```C++
std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
```
but I'm just going to keep the const where it is, as that's the case for other RPC calls. Sorry for the misinformation!

Reply:
I opened a separate pull request on this subject. Thanks for the nudge. https://github.com/bitcoin/bitcoin/pull/22787